### ISSUE TITLE
bug - copy module returns 'path' or 'dest' based on changed status - not consistent

### ISSUE THREAD

**Issue by halberom** [ansible/ansible#25431]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
template
copy

(template calls copy, I first spotted it when using template)

##### ANSIBLE VERSION
```
ansible-playbook 2.3.0.0
  config file = /Users/lynchg/git/gitlab.app.betfair/88882/ansible_arista/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 18 2016, 07:03:34) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

##### CONFIGURATION
NA

##### OS / ENVIRONMENT
Mac

##### SUMMARY
In the results returned from the template and copy modules, 'path' is defined when the destination is unchanged, but 'dest' is defined when it's changed.  This isn't consistent and makes following conditionals overly complicated.

##### STEPS TO REPRODUCE
https://gist.github.com/halberom/c2e5948a5811bd839a7f0bd31a344d9a

##### EXPECTED RESULTS
The same key names are used irrespective of changed results.

##### ACTUAL RESULTS
See the gist, output included

**Comment by ktosiek** [comment]
Affects 2.4, too: https://github.com/ansible/ansible/issues/31477
Edit: hmm, can't reproduce this on 2.3. But it is broken on 2.4.

**Comment by guzzijason** [comment]
Definitely see this in 2.4.2.0-1.el7 on CentOS. Was able to work around it using something like this:

```
- name: use instead of relying on inconsistent attributes from copy module
  set_fact: myvar="{{ copy_res.results[0].dest | default(copy_res.results[0].path) }}"
```

**Comment by nchepanov** [comment]
This appears to be fixed in the latest version
`ansible 2.6.0 (devel abb353290c) last updated 2018/05/14 16:36:46 (GMT -400)`

```
---
- hosts: all
  gather_facts: False
  connection: local
  vars:
    foo: bar
  tasks:
    - name: first time - dest is changed, 'dest' is returned in results
      template: src=test.j2 dest=/tmp/file
      register: result

    - debug: var=result.dest
    - debug: var=result.path

    - name: second time - nothing is changed, 'path' and `dest` is returned in results
      template: src=test.j2 dest=/tmp/file
      register: result

    - debug: var=result.dest
    - debug: var=result.path
```

Playbook execution output (redacted for brevity)
```
$ rm /tmp/file; ansible-playbook playbook.yml

<<<template the first time>>>

TASK [debug] 
ok: [127.0.0.1] => {
    "result.dest": "/tmp/file"
}

ok: [127.0.0.1] => {
    "result.path": "VARIABLE IS NOT DEFINED!"
}

 <<< teplate again >>>

TASK [debug] 
ok: [127.0.0.1] => {
    "result.dest": "/tmp/file"
}

TASK [debug]
ok: [127.0.0.1] => {
    "result.path": "/tmp/file"
}
```

**Comment by sivel** [comment]
@nikitos3000 thank you for testing and validating that this is now resolved.

I've confirmed that this was fixed in 0ac4722b29ee92be7130c6d8b07e8251127df6a2 and subsequently included in the 2.4.4 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
